controllers/operatorpolicy_controller.go

```diff
@@ -278,57 +404,42 @@ func (r *OperatorPolicyReconciler) updatePolicyStatus(
 func buildSubscription(
 	policy *policyv1beta1.OperatorPolicy,
-) *operatorv1alpha1.Subscription {
+) (*operatorv1alpha1.Subscription, error) {
 	subscription := new(operatorv1alpha1.Subscription)
 
-	subscription.SetGroupVersionKind(subscriptionGVK)
-	subscription.ObjectMeta.Name = policy.Spec.Subscription.Package
-	subscription.ObjectMeta.Namespace = policy.Spec.Subscription.Namespace
-	subscription.Spec = policy.Spec.Subscription.SubscriptionSpec.DeepCopy()
+	sub := make(map[string]interface{})
 
-	OpLog.Info("Creating the subscription", "kind", subscription.Kind,
-		"namespace", subscription.Namespace)
+	err := json.Unmarshal(policy.Spec.Subscription.Raw, &sub)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling subscription: %w", err)
+	}
 
-	return subscription
-}
+	ns, ok := sub["namespace"].(string)
+	if !ok {
+		return nil, fmt.Errorf("namespace is required in spec.subscription")
+	}
 
-// Sets the compliance of the policy
-func (r *OperatorPolicyReconciler) setCompliance(
-	ctx context.Context,
-	policy *policyv1beta1.OperatorPolicy,
-	compliance policyv1.ComplianceState,
-) {
-	policy.Status.ComplianceState = compliance
+	spec := new(operatorv1alpha1.SubscriptionSpec)
 
-	err := r.updatePolicyStatus(ctx, policy)
+	err = json.Unmarshal(policy.Spec.Subscription.Raw, spec)
 	if err != nil {
-		OpLog.Error(err, "error while updating policy status")
+		return nil, fmt.Errorf("error unmarshalling subscription: %w", err)
 	}
-}
-
-// buildOperatorGroup bootstraps the OperatorGroup spec defined in the operator policy
-// with the apiversion and kind in preparation for resource creation
-func buildOperatorGroup(
-	policy *policyv1beta1.OperatorPolicy,
-) *operatorv1.OperatorGroup {
-	operatorGroup := new(operatorv1.OperatorGroup)
 
-	// Create a default OperatorGroup if one wasn't specified in the policy
-	// Future work: Prevent creating multiple OperatorGroups in the same ns
-	if policy.Spec.OperatorGroup == nil {
-		operatorGroup.SetGroupVersionKind(operatorGroupGVK)
-		operatorGroup.ObjectMeta.SetName(policy.Spec.Subscription.Package + defaultOGSuffix)
-		operatorGroup.ObjectMeta.SetNamespace(policy.Spec.Subscription.Namespace)
-		operatorGroup.Spec.TargetNamespaces = []string{}
-	} else {
-		operatorGroup.SetGroupVersionKind(operatorGroupGVK)
-		operatorGroup.ObjectMeta.SetName(policy.Spec.OperatorGroup.Name)
-		operatorGroup.ObjectMeta.SetNamespace(policy.Spec.OperatorGroup.Namespace)
-		operatorGroup.Spec.TargetNamespaces = policy.Spec.OperatorGroup.Target.Namespace
-	}
+	subscription.SetGroupVersionKind(subscriptionGVK)
+	subscription.ObjectMeta.Name = spec.Package
+	subscription.ObjectMeta.Namespace = ns
+	subscription.Spec = spec
 
-	OpLog.Info("Creating the operator group", "kind", operatorGroup.Kind,
-		"namespace", operatorGroup.Namespace)
+	return subscription, nil
+}
 
-	return operatorGroup
+func opPolIdentifier(namespace, name string) depclient.ObjectIdentifier {
```

Comment:
Optional: This seems like a good candidate as a method on the OperatorPolicy struct.